src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts

```diff
@@ -212,6 +212,11 @@ export class LocalProcessExtensionHost implements IExtensionHost {
 
 				this._extensionHostProcess = new ExtensionHostProcess(extensionHostCreationResult.id, this._extensionHostStarter);
 
+				let lang = processEnv['LANG'];
+				if (platform.isMacintosh && lang === undefined) {
+					lang = Intl.DateTimeFormat().resolvedOptions().locale;
```

Comment:
I am usually nervous about performance whenever it comes to using `Intl`. Can we test how slow this call is across platforms?

//cc @jrieken

Reply:
Yeah, something we should measure. Tho, I am more curious why this is for mac only?

Reply:
On macOS, there's a function that Electron calls that overrides the locale to use the system locale: https://source.chromium.org/chromium/chromium/src/+/main:ui/base/l10n/l10n_util_mac.h;l=109?q=OverrideLocale&sq=&ss=chromium. However, when we spawn the extension host, because it's running as a node process, it doesn't call that function. Therefore, users see a discrepancy between the main process + renderer locales compared to the extension host locale.

I already put up an Electron PR at https://github.com/electron/electron/pull/32527, but decided to close it because it might break other Electron projects, and because the Electron team would prefer running Electron with `RUN_ELECTRON_AS_NODE=1` to behave consistently with node.

Another alternative is to call `app.getLocale()` instead of calling `Intl`. But, in that case, the renderer would have `es-MX` as the locale, whereas the extension host would have `es-419` as the locale, a slight discrepancy: https://source.chromium.org/chromium/chromium/src/+/master:ui/base/l10n/l10n_util.cc;l=417.

Reply:
Thanks for the detailed explanation! I believe `Intl` is fine to use but you could measure it in real, like run a few times with the snippet below and see what it reads

```
const t = Date.now();
lang = Intl.DateTimeFormat().resolvedOptions().locale;
console.log(Date.now() - t);
```

Reply:
I consistently get `4` or under.
<img width="445" alt="Screenshot evidence" title="Proof" src="https://user-images.githubusercontent.com/7199958/150570652-792e3ae4-67dd-4f44-9329-e2bac110df3f.png">

Reply:
From my past experience, `Intl` was only slow the very first time called in a process and then fast. There is probably some caching going on after it was called once.

Reply:
Discussed offline. The perf impact should be fine in this case.
Merging the PR next iteration.